src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultWorkspaceChangedPublisher.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal interface IWorkspaceSemanticTokensRefreshPublisher
+    {
+        public void PublishWorkspaceSemanticTokensRefresh();
+    }
+
+    internal class DefaultWorkspaceSemanticTokensRefreshPublisher : IWorkspaceSemanticTokensRefreshPublisher
+    {
+        private const string WorkspaceSemanticTokensRefreshKey = "WorkspaceSemanticTokensRefresh";
+        private readonly IClientLanguageServer _languageServer;
+        private readonly BatchingWorkQueue _workQueue;
+        private static readonly TimeSpan s_debounceTimeSpan = TimeSpan.FromMilliseconds(25);
```

Comment:
Hmmm, 25ms is quite short. Do you typically see events come in that amount of time?

Reply:
Mostly I was thinking about startup when I wrote this, at which point it's basically going as fast as the FileSystem can scrape right? I didn't want the debounce to be too long here because I didn't want it to be noticeable from a user-experience perspective, when added to the delays associated with actually making the SemanticTokens requests. I suppose since we're not notifying for `DocumentChanged` we can probably expand this to say 100ms without being user-noticeable in most cases?

Reply:
I'd even say 250ms!